### PR TITLE
Graduate PodSchedulingReadiness to stable

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind feature
/sig scheduling

#### What this PR does / why we need it:

Graduate PodSchedulingReadiness to stable

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/3521

#### Does this PR introduce a user-facing change?

```release-note
Graduated pod scheduling gates to general availability.
The `PodSchedulingReadiness` feature gate no longer has any effect, and the
`.spec.schedulingGates` field is always available within the Pod and PodTemplate APIs.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/3521-pod-scheduling-readiness
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
API changes only involve document updates.

**Comment by sftim** [comment]
Changelog suggestion
```diff
-Graduate PodSchedulingReadiness to stable
+Graduated pod scheduling gates to general availability.
+The `PodSchedulingReadiness` feature gate no longer has any effect, and the
+`.spec.schedulingGates` field is always available within the Pod and PodTemplate APIs.
```

**Comment by Huang-Wei** [comment]
cc @ahg-g

**Comment by ahg-g** [comment]
/lgtm

Thanks Wei!

**Comment by Huang-Wei** [comment]
@liggitt would you minding review the API parts (comments update only)?

File: pkg/scheduler/apis/config/testing/defaults/defaults.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 var PluginsV1 = &config.Plugins{
 	MultiPoint: config.PluginSet{
 		Enabled: []config.Plugin{
+			{Name: names.SchedulingGates},
```
Comment (liggitt) [inline pkg/scheduler/apis/config/testing/defaults/defaults.go:28]: does this ordering change modify behavior?

File: pkg/scheduler/apis/config/v1/default_plugins.go [hunk 2]
```diff
@@ -60,9 +61,6 @@ func getDefaultPlugins() *v1.Plugins {
 }
 
 func applyFeatureGates(config *v1.Plugins) {
-	if utilfeature.DefaultFeatureGate.Enabled(features.PodSchedulingReadiness) {
-		config.MultiPoint.Enabled = append(config.MultiPoint.Enabled, v1.Plugin{Name: names.SchedulingGates})
```
Comment (liggitt) [inline pkg/scheduler/apis/config/v1/default_plugins.go:64]: does moving this from the end of the list to the beginning of the list modify scheduler behavior?

**Comment by liggitt** [comment]
API comments lgtm, one question on the order change in the default multipoint scheduler plugins

File: pkg/scheduler/apis/config/v1/default_plugins.go (hunk 2 above)
Comment (Huang-Wei) [reply pkg/scheduler/apis/config/v1/default_plugins.go:64]: No, it doesn't. The plugins' order defined in MultiPoint only matter for those ones that belong to the same extension point. `SchedulingGates` is one and the only one plugin that implements the PreEnqueue extension point, so it doesn't intervene with other plugins, and hence doesn't change scheduler's behavior.

Moreover, PreEnqueue is the first step of the whole scheduling workflow, so placing its plugins in the head of Multipoint makes the code more readable, although it doesn't need to.

**Pull request merged**